async-nats/src/header.rs

```diff
@@ -44,6 +45,23 @@ use serde::{Deserialize, Serialize};
 /// # }
 /// ```
 
+#[derive(Clone, Debug, PartialEq)]
```

Comment:
My "final" target is to substitute the dynamic error `crate::Error` in public methods.

With this goal in mind, I looked at the code of nested methods where the error occurs. One of them is [crate::jetstream::kv::entry]() which uses the private [crate::Message::try_from]() which uses the private [crate::stream::parse_headers]().

This later beast parses both names and values and can return as `dyn crate::Error` both versions. That's why my first intention was to tell "these 2 errors are kinds of possible parsing-related problems that can be unioned".

But after you question I made a second look and recognized that in fact you're right, and my aggregation was a bit "premature" because the same error is raised not in parsing, but building context, where these 2 errors aren't necessarily related to each other.

That's why now I'm going to roll back here and instead of aggregating these 2 errors from the beginning, keep them separate, but provide the `crate::stream::ParseError` instead for all variants of the error that a `parse_headers` function should return. In that case aggregation should make more sense.

wdyt?